### PR TITLE
Graph-mode quantization for convolution from traced model

### PR DESCRIPTION
**Pull request by lly-zero-one**
In the PR, we enhance the graph-mode quantization for aten::_convolution, which could be generated from tracing path.

### CODE REVIEW

File: torch/csrc/jit/passes/quantization.cpp [hunk 1]
```diff
@@ -697,6 +698,41 @@ graph(%a_dequant, %w_quant, %b, %stride, %padding, %dilation, %groups):
   rewriter.runOnGraph(graph);
 }
 
+void insertPrepackUnpackForConvolution(std::shared_ptr<Graph>& graph) {
```
Comment (jerryzh168) [inline torch/csrc/jit/passes/quantization.cpp]: please change this to replacing `aten::_convolution` to `aten::conv2d` as discussed.

File: test/test_quantization.py [hunk 2]
```diff
@@ -817,15 +817,19 @@ def test_conv(self):
...
-        model_script = quantize_script(
-            torch.jit.script(conv_model_to_script),
-            qconfig_dict,
-            default_eval_fn,
-            [self.img_data],
-            inplace=False)
-        result_eager = model_eager(self.img_data[0][0])
-        result_script = model_script(self.img_data[0][0])
-        self.assertEqual(result_eager, result_script)
+        for trace in [True, False]:
```
Comment (ZolotukhinM) [inline test/test_quantization.py]: Nit: I'd rather trace and script before the loop and then iterate over two scripted models in the loop, i.e.:
```
traced = torch.jit.trace(conv_model_to_script, self.img_data[0][0])
scripted = torch.jit.script(conv_model_to_script)
for model_under_test in [traced, scripted]:
  ...
```
Also, `result_eager` can be computed once before the loop.

File: torch/csrc/jit/passes/quantization.cpp [hunk 3]
```diff
@@ -83,6 +83,7 @@ bool nodeQuantizable(Node* n) {
       "conv2d",
       "linear",
       "relu",
+      "_convolution",
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization.cpp]: How about we call `replaceConvolutionWithConv2d` in the very beginning, before we even inserted observers? In this case we won't need to extend the special handling of `conv2d` to `_convolution`. For instance, you've already forgotten to update one function: `isBiasOfConvOrLinear`.

File: torch/csrc/jit/passes/quantization.cpp [hunk 4]
```diff
@@ -697,6 +698,39 @@ graph(%a_dequant, %w_quant, %b, %stride, %padding, %dilation, %groups):
...
+  auto filter = [](const Match& match,
+                   const std::unordered_map<std::string, Value*>& vmap) {
+    const auto& match_vmap = match.values_map;
+    auto transposed_value = getIValue("transposed", match_vmap, vmap).value().toBool();
+    auto benchmark_value = getIValue("benchmark", match_vmap, vmap).value().toBool();
+    auto deterministic_value = getIValue("deterministic", match_vmap, vmap).value().toBool();
+    auto cudnn_enabled_value = getIValue("cudnn_enabled", match_vmap, vmap).value().toBool();
+    auto output_padding_value = getIValue("output_padding", match_vmap, vmap).value().toIntList();
+
+    if (!transposed_value && !benchmark_value && !deterministic_value && cudnn_enabled_value
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization.cpp]: Please clang-format the PR.

File: torch/csrc/jit/passes/quantization_patterns.h [hunk 5]
```diff
@@ -69,7 +69,6 @@ graph(%packed_params, %a_quant, %r_scale, %r_zero_point, %r_dtype):
     {matmul_with_bias, quantized_linear},
     {matmul_no_bias, quantized_linear_no_bias}
   };
-
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/quantization_patterns.h]: Nit: please try to avoid unnecessary changes.

**Review by ZolotukhinM — CHANGES_REQUESTED** [review]

**Review by ZolotukhinM — APPROVED** [review]
Looks good, thanks!

**Pull request closed without merging**